reactive/jenkins.py

```diff
@@ -230,3 +248,34 @@ def update_nrpe_config(nagios):
 def stop():
     service_stop("jenkins")
     status_set("maintenance", "Jenkins stopped")
+
+
+@when("reset.jenkins.storage")
+def reset_jenkins_storage():
+    status_set("maintenance", "Configuring Jenkins Storage")
+
+    jenkins_installed = get_state("apt.installed.jenkins")
+    if jenkins_installed:
+        service_stop('jenkins')
```

Comment:
Yes you could but I realized that in the case where storage was specified on the charm deploy this hook will fire before installing Jenkins. As it is now when Jenkins installs it populates the storage directory immediately and is ready for use. If you wait until after Jenkins is installed you have to restart Jenkins to repopulate the dir and possibly have to redo bootstrapping. It is cleaner to do it correctly from the beginning rather than watching the install restart Jenkins repeatedly and repeat steps for each dir so I thought it worth putting in this if statement.